src/lsdb/catalog/catalog.py

```diff
@@ -469,12 +465,11 @@ def join(
             if output_catalog_name is None:
                 output_catalog_name = self.hc_structure.catalog_info.catalog_name
 
-            new_catalog_info = dataclasses.replace(
-                self.hc_structure.catalog_info,
-                catalog_name=output_catalog_name,
-                ra_column=self.hc_structure.catalog_info.ra_column + suffixes[0],
-                dec_column=self.hc_structure.catalog_info.dec_column + suffixes[0],
-            )
+            new_catalog_info = self.hc_structure.catalog_info.model_copy()
```

Comment:
```suggestion
            new_catalog_info = self.hc_structure.catalog_info.model_copy(update=dict(
                catalog_name=output_catalog_name,
                ra_column=catalog_info.ra_column + suffixes[0],
                dec_column=catalog_info.dec_column + suffixes[0],
            ))
```
If you want to maintain the same kind of keyword-argument initialization of a new instance with Pydantic that you had with dataclasses, the above works.

Reply:
Ah, I'm seeing that my suggestion won't catch misspellings of field names in the way that `dataclasses.replace` does, so it's probably better to set individual fields the way you had been, in these cases.

Reply:
With `extra=True` in pydantic, there's not a TON of hope of catching misspellings. But I like the idea of copying and setting new field values all in a single call!

pylint isn't happy with the `dict()` and would prefer `{}`.

Reply:
> With `extra=True` in pydantic, there's not a TON of hope of catching misspellings. But I like the idea of copying and setting new field values all in a single call!
> 
> pylint isn't happy with the `dict()` and would prefer `{}`.

I didn't realize that!  Poking around, it appears that this preference is driven by performance.

https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/use-dict-literal.html